power_grid_model_c/power_grid_model/include/power_grid_model/main_core/output.hpp

```diff
@@ -451,6 +458,39 @@ output_result(Component const& transformer_tap_regulator, MainModelState<Compone
     return transformer_tap_regulator.get_null_sc_output();
 }
 
+// output voltage regulator
+template <std::derived_from<VoltageRegulator> Component, class ComponentContainer,
+          steady_state_solver_output_type SolverOutputType>
+    requires model_component_state_c<MainModelState, ComponentContainer, Component>
+constexpr auto output_result(Component const& voltage_regulator,
+                             MainModelState<ComponentContainer> const& state,
+                             MathOutput<std::vector<SolverOutputType>> const& math_output,
+                             Idx const obj_seq) {
+    using sym = typename SolverOutputType::sym;
+
+    Idx2D const gen_math_id = [&]() {
+        return state.topo_comp_coup->load_gen[obj_seq];
+    }();
+    if (gen_math_id.group != -1) {
+        // is voltage regulator always in same group as the generator it regulates?
```

Comment:
yes because the group represents the connected component (in the graph-theoretical sense) it belongs to. That is, if you have 2 distinct subgrids that are not connected to eachother, then you will have 2 topological groups, and therefore also 2 math solvers (one for each subgrid).

As a special case, if a subgrid does not contain a source, the subgrid is considered disconnected, will not have an associated math solver, and therefore, the `group` for all components in that subgrid will be `-1`

if the groups are not equal, there's a bug in the code concerning topology